### PR TITLE
Prepare 14 projects to work in Containers

### PR DESCRIPTION
**Pull request by Sinerum**
Adds containers with the necesarry dependencies to build the latest Commit for the follwing projects:
bison, curl, htop, libjpeg_turbo, libssh, libxml2, lrzip, lz4, grep,  gzip, opus, openssl, poppler, tmux, xz,
Closes se-sic/VaRA#457,
Closes se-sic/VaRA#458

Also adds a container for gawk but something still does not work there

### CODE REVIEW

**Review by boehmseb — APPROVED** [review]
Looks good apart from the unused imports reported by pylint.

File: varats/varats/projects/c_projects/libssh.py [hunk 1]
```diff
@@ -90,15 +108,27 @@ def __compile_cmake(self) -> None:
                 bb.watch(cmake)("-G", "Unix Makefiles", "..")
 
             bb.watch(make)("-j", get_number_of_jobs(bb_cfg()))
-
+        with local.cwd(libssh_source):
```
Comment (vulder) [inline varats/varats/projects/c_projects/libssh.py:107]:
```suggestion

        with local.cwd(libssh_source):
```

**Review by vulder — CHANGES_REQUESTED** [review]
Can you please fix the open linter warnings, otherwise, this looks fine to me.

File: varats/varats/projects/c_projects/libssh.py [hunk 2]
```diff
@@ -3,17 +3,22 @@
 
 import benchbuild as bb
 from benchbuild.utils.cmd import make, cmake, mkdir
+from benchbuild.utils.revision_ranges import (
+    block_revisions,
+    GoodBadSubgraph
+)
```
Comment (vulder) [inline varats/varats/projects/c_projects/libssh.py]:
```suggestion
from benchbuild.utils.revision_ranges import block_revisions, GoodBadSubgraph
```

File: varats/varats/projects/c_projects/libssh.py [hunk 3]
```diff
@@ -3,17 +3,22 @@
...
+from varats.containers.containers import get_base_image, ImageBase
 from varats.paper_mgmt.paper_config import project_filter_generator
 from varats.project.project_util import (
     ProjectBinaryWrapper,
     get_all_revisions_between,
     wrap_paths_to_binaries,
     get_local_project_git_path,
     BinaryType,
-    verify_binaries,
+    verify_binaries
```
Comment (vulder) [inline varats/varats/projects/c_projects/libssh.py]:
```suggestion
    verify_binaries,
```

File: varats/varats/projects/c_projects/libvpx.py [hunk 4]
```diff
@@ -34,6 +35,10 @@ class Libvpx(bb.Project):  # type: ignore
         )
     ]
 
+    CONTAINER = get_base_image(
+        ImageBase.DEBIAN_10
+    ).run('apt', 'install', '-y', 'yasm')
```
Comment (vulder) [inline varats/varats/projects/c_projects/libvpx.py]:
```suggestion
    CONTAINER = get_base_image(ImageBase.DEBIAN_10
                              ).run('apt', 'install', '-y', 'yasm')
```

File: varats/varats/projects/cpp_projects/poppler.py [hunk 5]
```diff
@@ -3,9 +3,14 @@
 
 import benchbuild as bb
 from benchbuild.utils.cmd import cmake, make
+from benchbuild.utils.revision_ranges import (
+    block_revisions,
+    RevisionRange,
+)
```
Comment (vulder) [inline varats/varats/projects/cpp_projects/poppler.py]:
```suggestion
from benchbuild.utils.revision_ranges import block_revisions, RevisionRange
```

**Review by vulder — APPROVED** [review]
Fixes pre-commit errors

@Sinerum If you install pre-commit on your machine, these errors will be fixed automatically before creating an commit.

**Pull request merged**